cmd/kraft/kraft.go

```diff
@@ -85,6 +90,92 @@ func New() *cobra.Command {
 	return cmd
 }
 
+func (k *Kraft) PersistentPre(cmd *cobra.Command, args []string) error {
+	if config.G[config.KraftKit](cmd.Context()).NoCheckUpdates {
+		return nil
+	}
+
+	client := &http.Client{}
+	path := "https://get.kraftkit.sh/latest.txt"
+
+	get, err := http.NewRequest("GET", path, nil)
+	if err != nil {
+		log.G(cmd.Context()).WithError(err).Debug("failed to create request to get latest version")
+		return nil
+	}
+
+	get.Header.Set("User-Agent", "kraftkit/"+kitversion.Version())
+	log.G(cmd.Context()).WithFields(logrus.Fields{
+		"url":    path,
+		"method": "GET",
+	}).Trace("http")
+
+	resp, err := client.Do(get)
+	if err != nil {
+		log.G(cmd.Context()).WithError(err).Debug("failed to get latest version")
+		return nil
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		log.G(cmd.Context()).WithError(err).Debug("failed to get latest version")
+		return nil
+	}
+
+	contents, err := io.ReadAll(resp.Body)
+	if err != nil {
+		log.G(cmd.Context()).WithError(err).Debug("failed to read latest version")
+		return nil
+	}
+
+	// Latest version parsing: 0.5.0-rc1 -> ["0", "5", "0"]
+	latestVersionToParse := strings.Split(string(contents), "-")[0]
+	latestVersionNumbers := strings.Split(latestVersionToParse, ".")
```

Comment:
Did you try https://pkg.go.dev/github.com/blang/semver ?

Reply:
switched